subsys/emds/emds_new.c

```diff
@@ -169,13 +170,93 @@ uint32_t emds_store_time_get(void)
 	return 0;
 }
 
+static uint8_t *emds_entry_memory_get(struct emds_data_entry *entry)
+{
+	STRUCT_SECTION_FOREACH(emds_entry, ch) {
+		if (ch->id == entry->id) {
+			entry->length = MIN(ch->len, entry->length);
+			return ch->data;
+		}
+	}
+
+	struct emds_dynamic_entry *ch;
+
+	SYS_SLIST_FOR_EACH_CONTAINER(&emds_dynamic_entries, ch, node) {
+		if (ch->entry.id == entry->id) {
+			entry->length = MIN(ch->entry.len, entry->length);
+			return ch->entry.data;
+		}
+	}
+
+	LOG_WRN("Entry with ID %u not found", entry->id);
+	return NULL;
+}
+
+static int emds_read_data(const struct flash_area *fa, struct emds_snapshot_metadata *metadata)
+{
+	struct emds_data_entry entry;
+	uint32_t data_addr = metadata->data_instance_addr;
+	int32_t data_len = metadata->data_instance_len;
+	uint8_t *data_buf;
+	int rc;
+
+	while (data_len > 0) {
+		rc = flash_area_read(fa, data_addr, &entry, sizeof(entry));
+		if (rc) {
+			LOG_ERR("Failed to read data entry: %d", rc);
+			return -EIO;
+		}
+
+		data_addr += sizeof(entry);
+		data_len -= sizeof(entry);
+
+		data_buf = emds_entry_memory_get(&entry);
+
+		if (data_buf) {
+			rc = flash_area_read(fa, data_addr, data_buf, entry.length);
+			if (rc) {
+				LOG_ERR("Failed to read data for entry ID %u: %d", entry.id, rc);
+				return -EIO;
+			}
+		}
+
+		data_addr += entry.length;
+		data_len -= entry.length;
+	}
+
+	return 0;
+}
+
 int emds_load(void)
 {
+	struct emds_snapshot_metadata metadata = {0};
+	int freshest_partition_idx = -1;
+
 	if (!emds_initialized) {
 		return -ECANCELED;
 	}
 
-	return 0;
+	for (int i = 0; i < PARTITIONS_NUM_MAX; i++) {
+		if (emds_flash_scan_partition(&partition[i], &metadata)) {
+			LOG_ERR("Failed to scan partition %d:", i);
```

Comment:
Should this be a LOG_DBG (LOG_WRN) as possible no error code will be supplied?

Reply:
Hm... This seems real error that is raised from driver. I'd prefer to keep it as LOG_ERR. EMDS doesn't stop scanning here, only because there is one more partition that might work normally.